src/sortStudents.ts

```diff
@@ -1,16 +1,59 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grade',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function averageGradeCalc(grades: number[]): number {
+  return grades
+    .reduce((grade: number, sum: number) => grade + sum) / grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copyStudents = [...students];
+  let sign = 1;
+
+  function sortStudentsFunction(a: Student, b: Student): number {
+    switch (sortBy) {
+      case SortType.Name:
+      case SortType.Surname:
+        return sign * (a[sortBy].localeCompare(b[sortBy]));
+
+      case SortType.AverageGrade:
+        return sign
+          * (averageGradeCalc(a.grades) - averageGradeCalc(b.grades));
+
+      case SortType.Age:
+      case SortType.Married:
+        return sign * (+a[sortBy] - +b[sortBy]);
```

Comment:
avoid such expressions `- +`. Use [Number](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number) convert 
```suggestion
        return sign * (+a[sortBy] - Number(b[sortBy]));
```